### PR TITLE
feat(AIP-154): add more tests around resource freshness validation

### PR DESCRIPTION
**Pull request by thall**
This PR adds more tests to ensure [AIP-154](https://google.aip.dev/154) is correctly fulfilled.

See each commit messages to see what is verified and not. 

Same as https://github.com/einride/protoc-gen-go-aip-test/pull/240 but structured in another way, hopefully this is easier to review.

### CODE REVIEW

File: internal/aiptest/deletion/already_deleted.go [hunk 1]
```diff
@@ -29,15 +29,18 @@ var alreadyDeleted = suite.Test{
...
-			Method:   deleteMethod,
-			Name:     "created.Name",
-		}.Generate(f, "_", "err", ":=")
+			Resource:    scope.Resource,
+			Method:      deleteMethod,
+			ResourceVar: "created",
+		}.Generate(f, "deleted", "err", ":=")
 		f.P(ident.AssertNilError, "(t, err)")
+		// prevent no usage error
+		f.P("_ = deleted")
```
Comment (thall) [inline internal/aiptest/deletion/already_deleted.go:38]: we maybe generating some unnecessary code but its way simple than what I did [here](https://github.com/einride/protoc-gen-go-aip-test/pull/240/commits/ccaffdce07ae5333b069c833adcd61d7114cd156#diff-949601ae252056096b249929850f8a077a90d5f51fbbd5547dd6d28362b9a706R31-R39)

**Review by odsod — APPROVED** [review]
Awesome to have these tests!!

File: internal/aiptest/deletion/already_deleted.go (hunk 1 above)
Comment (odsod) [reply internal/aiptest/deletion/already_deleted.go:38]: hehe yeah all good!

**Review by ericwenn — APPROVED** [review]

File: internal/aiptest/update/etag.go [hunk 2]
```diff
@@ -39,3 +39,35 @@ var etagMismatch = suite.Test{
...
+			f.P("parent := ", ident.FixtureNextParent, "(t, false)")
+			f.P("created := fx.create(t, parent)")
+		} else {
+			f.P("created := fx.create(t)")
+		}
+		updateMethod, _ := util.StandardMethod(scope.Service, scope.Resource, aipreflect.MethodTypeUpdate)
+		util.MethodUpdate{
+			Resource: scope.Resource,
+			Method:   updateMethod,
+			Parent:   "parent",
```
Comment (thall) [inline internal/aiptest/update/etag.go:65]: thanks @ericwenn , well spotted!

**Pull request merged**